crypto/fipsmodule/ec/make_tables.go

```diff
@@ -341,6 +412,35 @@ func bigIntToU64s(curve elliptic.Curve, n *big.Int) []uint64 {
 	return ret
 }
 
+// Convert big int to an array of 58-bit digits.
+// This is needed for P-521 Fiat-crypto implementation.
+func bigIntToU58s(curve elliptic.Curve, n *big.Int) []uint64 {
+	words := (curve.Params().BitSize + 57) / 58
+	ret := make([]uint64, words)
+	mask := big.NewInt((1 << 58) - 1)
+	tmp := new(big.Int).Set(n)
+	for i := 0; i < words; i++ {
```

Comment:
Should the stopping condition be tmp == 0 as an additional check on the correct calculation of `words`? Alternatively, we can check after the loop that tmp =0.

Reply:
I think there is no reason for additional checks here, there are no checks in the existing functions and the script is used only for generating the table. Any mistake here and the table would be wrong, which means the unit tests for the curve would fail.